### ISSUE TITLE
Return custom class

### ISSUE THREAD

**Issue by Ryanauger95** [Vincit/objection.js#1578]
This isn't a bug, this is just a question. Absolutely loving objection btw.

So, any request I'm doing, from simple:
```
    const device = await DeviceModel.query()
      .findById(deviceId)
```
to more advanced:
```
    const device = await DeviceModel.query()
      .findById(request.deviceId)
      .select("*")
      .leftJoinRelation({
        next_app_version: true
      })
      .eager("location_operating_hours");
```
returns a ```Model[]``` object. But I need to access properties of the object (for example, ```location_id```, so for TS to compile, i need to cast to an ```any``` object.  Is there any way that I can return a more specific object, or convert my object into a class that I recognize? It doesn't have to be _through_ objection, I can write some helper functions, I'm just curious what the best practice is for this. One reason that it would be very helpful is if a field in the database ever changes names, or something is misspelled, I'd like to throw an error.

**Comment by koskimas** [comment]
The return type should be the model class, not `Model`. There may be a bug in the old 1.x typings if you get `Model[]` as the type.

**Comment by Ryanauger95** [comment]
Assuming it returned DeviceModel instead , would you just use getters to retrieve certain properties?

**Comment by Ryanauger95** [comment]
Or is there any other way that you would suggest?

**Comment by Ryanauger95** [comment]
Using 1.6.11 btw.

**Comment by koskimas** [comment]
The model class should have it's properties defined as properties, shouldn't it? No need for getters.

**Comment by koskimas** [comment]
I'm not able to reproduce the `Model[]` return value for anything other than queries that have `*joinRelation` call in them. It seems the the `*joinRelation` typings have a bug. Are you absolutely sure this happens for all your queries as you said in the first message?

**Comment by Ryanauger95** [comment]
@koskimas my mistake. Returns Device Model in query w/o relation.

And when I talk about properties, I should have said records. When I query a database, say using findOne, and it returns a record and I try to access a column, I get the error:
```
const tmp = await DeviceModel.query().findById(deviceId)
console.log(tmp.id)
 --- Property 'id' does not exist on type 'DeviceModel'.
```

So, is there any way that I could tell typescript that `id` does exist ? As well as the other fields? I believe that using jsonSchema should allow me to filter out any issues, and I should quickly get errors if I ever change a database column name with that, but still I end up casting to `any` whenever I access a field.

**Comment by koskimas** [comment]
You should define them as properties for the `DeviceModel` class because that's what you are trying to access:

```js
class DeviceModel extends Model {
  id!: number
  someOtherProp!: string
}
```

**Comment by koskimas** [comment]
The `Model[]` return value has been fixed in 2.0.
